File: src/test/java/no/nav/sosialhjelp/soknad/consumer/arbeidsforhold/dto/ArbeidsforholdDtoTest.java
package no.nav.sosialhjelp.soknad.consumer.arbeidsforhold.dto;

import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.DeserializationFeature;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
import org.apache.commons.io.IOUtils;
import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.nio.charset.StandardCharsets;
import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

class ArbeidsforholdDtoTest {

    @Test
    void skalDeserialisereArbeidsforholdResponse() throws IOException {
        var objectMapper = new ObjectMapper()
                .enable(DeserializationFeature.ACCEPT_SINGLE_VALUE_AS_ARRAY)
                .registerModule(new JavaTimeModule());

        var resourceAsStream = ClassLoader.getSystemResourceAsStream("arbeidsforhold/aaregResponse.json");
        assertThat(resourceAsStream).isNotNull();
        var jsonString = IOUtils.toString(resourceAsStream, StandardCharsets.UTF_8);

        var arbeidsforholdDtoList = objectMapper
                .readValue(jsonString, new TypeReference<List<ArbeidsforholdDto>>() {
                });

        var dto = arbeidsforholdDtoList.get(0);
        assertThat(dto.getAnsettelsesperiode().getPeriode().getFom()).hasToString("2014-07-01");
        assertThat(dto.getAnsettelsesperiode().getPeriode().getTom()).hasToString("2015-12-31");
        assertThat(dto.getArbeidsavtaler()).hasSize(1);
        assertThat(dto.getArbeidsavtaler().get(0).getStillingsprosent()).isEqualTo(49.5);
        assertThat(dto.getArbeidsforholdId()).isEqualTo("abc-321");
        assertThat(dto.getArbeidsgiver()).isExactlyInstanceOf(OrganisasjonDto.class);
        assertThat(dto.getArbeidstaker().getOffentligIdent()).isEqualTo("31126700000");
        assertThat(dto.getNavArbeidsforholdId().longValue()).isEqualTo(123456L);
    }

}

File: src/test/kotlin/no/nav/sosialhjelp/soknad/arbeid/dto/ArbeidsforholdDtoTest.kt
package no.nav.sosialhjelp.soknad.arbeid.dto

import com.fasterxml.jackson.databind.DeserializationFeature
import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
import com.fasterxml.jackson.module.kotlin.readValue
import org.apache.commons.io.IOUtils
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import java.nio.charset.StandardCharsets

internal class ArbeidsforholdDtoTest {

    @Test
    fun skalDeserialisereArbeidsforholdResponse() {
        val objectMapper = jacksonObjectMapper()
            .enable(DeserializationFeature.ACCEPT_SINGLE_VALUE_AS_ARRAY)
            .disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES)
            .registerModule(JavaTimeModule())

        val resourceAsStream = ClassLoader.getSystemResourceAsStream("arbeidsforhold/aaregResponse.json")
        assertThat(resourceAsStream).isNotNull

        val jsonString = IOUtils.toString(resourceAsStream, StandardCharsets.UTF_8)
        val arbeidsforholdDtoList = objectMapper.readValue<List<ArbeidsforholdDto>>(jsonString)
        val dto = arbeidsforholdDtoList[0]
        assertThat(dto.ansettelsesperiode?.periode?.fom).hasToString("2014-07-01")
        assertThat(dto.ansettelsesperiode?.periode?.tom).hasToString("2015-12-31")
        assertThat(dto.arbeidsavtaler).hasSize(1)
        assertThat(dto.arbeidsavtaler!![0].stillingsprosent).isEqualTo(49.5)
        assertThat(dto.arbeidsforholdId).isEqualTo("abc-321")
        assertThat(dto.arbeidsgiver).isExactlyInstanceOf(OrganisasjonDto::class.java)
        assertThat(dto.arbeidstaker?.offentligIdent).isEqualTo("31126700000")
    }
}
